### PR TITLE
onfido/add_sup_doc

### PR DESCRIPTION
**Pull request by chengyang-binary**

### CODE REVIEW

File: lib/WebService/Async/Onfido.pm [hunk 1]
```diff
@@ -1034,6 +1034,52 @@ sub countries_list {
...
+Returns the supported_documents_list for the country. 
+
+=cut
+
+sub supported_documents_for_country {
+    my ($self, $country_code) = @_;
+
+    my %country_to_doc = map { $_->{country_code} => $_ } @{supported_documents_list()};
+
+    return $country_to_doc{$country_code}->{doc_types_list};
```
Comment (raunakkathuria) [inline lib/WebService/Async/Onfido.pm]:
```suggestion
    return $country_to_doc{$country_code}->{doc_types_list} // [];
```

File: lib/WebService/Async/Onfido.pm [hunk 2]
```diff
@@ -1034,6 +1034,52 @@ sub countries_list {
...
+
+sub supported_documents_list {
+    my $path = Path::Tiny::path(__DIR__)->parent(3)->child('share/supported_documents.json');
+    $path = Path::Tiny::path(
+        File::ShareDir::dist_file(
+            'WebService-Async-Onfido',
+            'supported_documents.json'
+        )
+    ) unless $path->exists;
+    my $supp_docs = decode_json_text($path->slurp_utf8);
```
Comment (raunakkathuria) [inline lib/WebService/Async/Onfido.pm]: better to use full name `$supported_documents`

File: lib/WebService/Async/Onfido.pm [hunk 3]
```diff
@@ -1034,6 +1034,52 @@ sub countries_list {
...
+=head2 supported_documents_for_country
+
+Returns the supported_documents_list for the country. 
+
+=cut
+
+sub supported_documents_for_country {
+    my ($self, $country_code) = @_;
+
+    my %country_to_doc = map { $_->{country_code} => $_ } @{supported_documents_list()};
```
Comment (raunakkathuria) [inline lib/WebService/Async/Onfido.pm]: I think better name would be `country_details`

File: lib/WebService/Async/Onfido.pm (hunk 2 above)
Comment (chengyang-binary) [reply lib/WebService/Async/Onfido.pm]: changed

File: lib/WebService/Async/Onfido.pm (hunk 1 above)
Comment (chengyang-binary) [reply lib/WebService/Async/Onfido.pm]: changed

File: lib/WebService/Async/Onfido.pm (hunk 3 above)
Comment (chengyang-binary) [reply lib/WebService/Async/Onfido.pm]: changed

File: lib/WebService/Async/Onfido.pm [hunk 4]
```diff
@@ -1034,6 +1033,52 @@ sub countries_list {
...
+
+sub supported_documents_list {
+    my $path = Path::Tiny::path(__DIR__)->parent(3)->child('share/supported_documents.json');
+    $path = Path::Tiny::path(
+        File::ShareDir::dist_file(
+            'WebService-Async-Onfido',
+            'supported_documents.json'
+        )
+    ) unless $path->exists;
+    my $supported_documents = decode_json_text($path->slurp_utf8);
```
Comment (chylli-binary) [inline lib/WebService/Async/Onfido.pm]: Can we cache this result ?

Comment (chengyang-binary) [reply lib/WebService/Async/Onfido.pm]: the json file is actually added in `bom-config`, this is just for public usage

Comment (chylli-binary) [reply lib/WebService/Async/Onfido.pm]: Sorry I can't get your idea.
I mean, can I do like this ?
```state $supported_documents = decode_json_text($path->slurp_utf8)```

**Pull request merged**